pandas/tests/dtypes/test_inference.py

```diff
@@ -607,6 +607,20 @@ def test_convert_infs(self):
         result, _ = lib.maybe_convert_numeric(arr, set(), False)
         assert result.dtype == np.float64
 
+    @pytest.mark.parametrize(
+        "arr",
+        [
+            np.array([1 + 2j, 2 + 3j, 3 + 4j, 4 + 5j], dtype="O"),
+            np.array([1 + 2j, 2 + 3j, None, 4 + 5j], dtype="O"),
+            np.array([1 + 2j, 2, 3.14, np.complex128(np.nan)], dtype="O"),
+            np.array([2, 1 + 2j, 3.14, np.complex128(np.nan)], dtype="O"),
+        ],
+    )
+    def test_convert_complexes(self, arr):
+        result, _ = lib.maybe_convert_numeric(arr, set())
+        expected = arr.astype(np.complex128)
+        tm.assert_numpy_array_equal(result, expected)
```

Comment:
is this a bugfix?  if so, can you either a) separate PR or b) comment (in GH) on where the relevant edits are